### DISCUSSION TITLE
Warning: Some installed kegs have no formulae (pycparser)

### DISCUSSION THREAD

**Discussion by chrisfinazzo** [Homebrew/discussions#4353] [Everyday usage] [answered]
### Output of `brew config`

```shell
Chris in [~] % brew config
HOMEBREW_VERSION: 4.0.9-57-g86c518e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 86c518e2d7180b054b7b215dad87f2780379d537
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 4fbf693010429a62c6d6a807750262508000736a
Core tap last commit: 6 weeks ago
Core tap branch: master
Core tap JSON: 24 Mar 02:33 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.LQJpgzZ7dN/org.xquartz:0
HOMEBREW_EDITOR: /usr/local/bin/bbedit
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 14.0.0 build 1400
Git: 2.40.0 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6-x86_64
CLT: 14.0.0.0.1.1661618636
Xcode: 14.1
```

### Output of `brew doctor`

```shell
Chris in [~] % brew doctor

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  pycparser
```

### Description of issue

_Note: Context is in Homebrew Core [#106902][]._

[#106902]: https://github.com/Homebrew/homebrew-core/issues/106902

This was the result of an upgrade to OCRmyPDF. Initially I suspected it might be deprecated, so I went about looking for a replacement. Finding nothing suitable, I attempted to reinstall the currently available formula, which produced the error shown above.

Is there a tap which can provide a replacement?

```shell
Chris in [~] % brew info pycparser
==> pycparser: stable 2.21 (bottled)
C parser in Python
https://github.com/eliben/pycparser
/usr/local/Cellar/pycparser/2.21 (46 files, 656.7KB) *
  Poured from bottle using the formulae.brew.sh API on 2023-03-23 at 22:37:16
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/pycparser.rb
License: BSD-3-Clause
==> Dependencies
Required: python@3.11 ✔
==> Analytics
install: 58 (30 days), 58 (90 days), 58 (365 days)
install-on-request: 0 (30 days), 0 (90 days), 0 (365 days)
build-error: 0 (30 days)
```

**Comment by carlocab** [comment, marked as answer]
Can you try doing
```
brew untap homebrew/core
```
and see if it helps? Note that this will delete your local homebrew/core repo. You may wish to back it up if you have changes you'd like to save.

**Reply by chrisfinazzo** [reply]
Ah, that does it.

I was quite confused at first when I read in the 4.0.0 [release notes][] that you can untap `homebrew/core` and that everything doesn't implode.

On the occasion I do submit an update to a formula (or cask, for that matter), I usually do it through a branch in my local clones, so those should continue to work.

[release notes]: https://brew.sh/2023/02/16/homebrew-4.0.0/

**Comment by maartenbosteels** [comment]
I have the same issue (after installing awscli).
What will ``brew untap homebrew/core`` do ?

**Reply by ZhongRuoyu** [reply]
It removes the local copy of the tap repository (see [manpage](https://docs.brew.sh/Manpage#untap---force-tap-)). This is fine if you do not work on the repository, because the default option since brew 4.0.0 is to fetch formulae information from the JSON API (see [4.0.0 release notes](https://brew.sh/2023/02/16/homebrew-4.0.0/)). If you run `brew doctor`, you should see that the `homebrew/core` tap has not been updated for a while.

**Comment by cclauss** [comment]
https://github.com/Homebrew/homebrew-core/blob/master/Formula/pycparser.rb says:
>  ___Cannot retrieve latest commit at this time.___
